lib/tasks/plugin_test.rake

```diff
@@ -0,0 +1,79 @@
+require File.expand_path('../../config/environment', File.dirname(__FILE__))
+
+desc 'Run a single or multiple plugins tests - called via rake plugin:test[plugin_name,plugin_name]'
+task 'plugin:test' => [:environment] do |t, args|
+  plugin_names = args.extras
+    
+  puts "You must specify the name of at least one plugin (e.g. rake test:plugin['my_plugin'])" if plugin_names.empty?
+
+  plugins = plugin_names.collect do |plugin_name|
+    plugin = Foreman::Plugin.find(plugin_name)
+    unless plugin
+      puts "No plugin found for #{plugin_name}. Availble plugins are:"
+      Rake::Task['plugins'].invoke
+      exit 1
+    end
+    plugin
+  end
+
+  plugins.each do |plugin|
+    Rake::Task["test:units"].enhance do
+      test_task = plugin.test_task.nil? ? "plugin:test:#{plugin.name}" : plugin.test_task
+      Rake::Task[test_task].invoke
+    end
+  end
+
+  Rake::Task['test:units'].invoke
```

Comment:
Why does this use test:units rather than test?

The Jenkins one below uses jenkins:unit (singular!) which runs all tests, but test:units would only ever run units.